src/instantiator.js

```diff
@@ -22,6 +22,23 @@ function isPrimitive(obj) {
 }
 
 /**
+ * Checks whether a property is on required array.
+ * @param property - the property to check.
+ * @param requiredArray - the required array
+ * @returns {boolean}
+ */
+function isPropertyRequired(property, requiredArray) {
+  var found = false;
+  var i = 0;
+  requiredArray = requiredArray || [];
+  while (i < requiredArray.length && !found) {
```

Comment:
Well, you are right. It is more clear with a `forEach`  I only use a `while` loop to be more efficient (stop the loop when it founds the required property).